Assignment 05
Note: You will be using Pandas for this assignment.
1.	Explore College Scorecard website - https://collegescorecard.ed.gov/
2.	Explore College Scorecard Data web page - https://collegescorecard.ed.gov/data/
3.	Download and read the documentation - https://collegescorecard.ed.gov/assets/FullDataDocumentation.pdf
4.	Download the dataset - https://ed-public-download.app.cloud.gov/downloads/CollegeScorecard_Raw_Data.zip (265MB Zip File)
5.	Unzip the file into a folder in your personal computer (there are many files, one for each school year)
6.	Download and read (use naked eyes, no coding) the data dictionary file - https://collegescorecard.ed.gov/assets/CollegeScorecardDataDictionary.xlsx
7. Find out the definition of the above 9 variables and document the definitions in the Jupyter Notebook in a Markdown cell so that the readers of your notebook will know the meaning of these variables.
8. Make sure your notebook file is well organized using Markdown and code comments



### Upload the 2017-2018 school year file "MERGED2017_18_PP.csv" to Colab/Deepnote or any Cloud environment of your choice (you can use your local environment too) and create a Jupyter notebook to explore the dataset:
- how many colleges or rows and variables or columns?
-	Look at the first 5 colleges, last 5 colleges, random sample of 5 colleges

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("MERGED2017_18_PP.csv")
print(df)

        UNITID    OPEID  ...  FTFTPCTFLOAN_POOLED_SUPP SCUGFFN_POOLED
0       100654   100200  ...                       NaN            NaN
1       100663   105200  ...                       NaN            NaN
2       100690  2503400  ...                       NaN            NaN
3       100706   105500  ...                       NaN            NaN
4       100724   100500  ...                       NaN            NaN
...        ...      ...  ...                       ...            ...
7107  48312404   108163  ...                       NaN            NaN
7108  48387801  4223701  ...                       NaN            NaN
7109  48400201  4228101  ...                       NaN            NaN
7110  48778201  4221501  ...                       NaN            NaN
7111  49005401  4182601  ...                       NaN            NaN

[7112 rows x 1986 columns]


/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2718: DtypeWarning: Columns (1729) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


### There are 7112 rows and 1986 columns for this data frame

In [3]:
# The first 5 colleges
df.head(5)

UNITID    OPEID  ...  FTFTPCTFLOAN_POOLED_SUPP SCUGFFN_POOLED
0  100654   100200  ...                       NaN            NaN
1  100663   105200  ...                       NaN            NaN
2  100690  2503400  ...                       NaN            NaN
3  100706   105500  ...                       NaN            NaN
4  100724   100500  ...                       NaN            NaN

[5 rows x 1986 columns]

In [4]:
# The last five colleges
df.tail(5)

UNITID    OPEID  ...  FTFTPCTFLOAN_POOLED_SUPP SCUGFFN_POOLED
7107  48312404   108163  ...                       NaN            NaN
7108  48387801  4223701  ...                       NaN            NaN
7109  48400201  4228101  ...                       NaN            NaN
7110  48778201  4221501  ...                       NaN            NaN
7111  49005401  4182601  ...                       NaN            NaN

[5 rows x 1986 columns]

In [5]:
# A random sample of 5
df.sample(5)

UNITID    OPEID  ...  FTFTPCTFLOAN_POOLED_SUPP SCUGFFN_POOLED
3391    222099   494700  ...                       NaN            NaN
4284    377722  2343400  ...                       NaN            NaN
4304    381352  3686300  ...                       NaN            NaN
6985  37727201  3009401  ...                       NaN            NaN
510     125471   859600  ...                       NaN            NaN

[5 rows x 1986 columns]

In [6]:
df1 = df

### Since there are too many variables to explore, many of which are not interesting. Create a smaller data frame with the following 9 variables:
- UNITID, INSTNM, CITY, STABBR, ZIP, ADM_RATE, UGDS, TUITIONFEE_IN, MN_EARN_WNE_P10

- Below are the variables and its explanation:
    - **UNITID** - This is the unit ID for the institution. Its API data type is an integer.
    - **INSTNM** - This is the institution name. Its API data type is autocomplete.
    - **CITY** - This is the city of the institution. Its API data type is autocomplete.
    - **STABBR** - This is state post code. Its API data type is string.
    - **ADM_RATE** - This is the admission rate. Its API data type is string.
    - **UGDS** - This is the enrollment of undergraduate certificate/ degree seeking students. Its API data type is integer.
    - **TUITION_IN** - This is the in-state tuition and fees integer tuition fee. Its API data type is integer.
    - **MN_EARN_WNE_P10** - This is mean earning of students working and not enrolled 10 years after entry. Its API data type is integer.


In [7]:
# The columns to be kept
df2 = df1[["UNITID", "INSTNM", "CITY", "STABBR", "ZIP", "ADM_RATE", "UGDS", "TUITIONFEE_IN", "MN_EARN_WNE_P10"]]

###Explore the smaller data frame
- Look at the first 10 colleges, last 10 colleges, random sample of 10 colleges
-	Find and display UMBC, Univ. of Maryland College Park, Loyola College Maryland, Johns Hopskins
- Calculate the min, max, and average in-state tution of the above four institutions.


In [8]:
df2.head(10)

UNITID                               INSTNM  ... TUITIONFEE_IN MN_EARN_WNE_P10
0  100654             Alabama A & M University  ...        9857.0             NaN
1  100663  University of Alabama at Birmingham  ...        8328.0             NaN
2  100690                   Amridge University  ...        6900.0             NaN
3  100706  University of Alabama in Huntsville  ...       10280.0             NaN
4  100724             Alabama State University  ...       11068.0             NaN
5  100751            The University of Alabama  ...       10780.0             NaN
6  100760    Central Alabama Community College  ...        4440.0             NaN
7  100812              Athens State University  ...           NaN             NaN
8  100830      Auburn University at Montgomery  ...        8020.0             NaN
9  100858                    Auburn University  ...       10968.0             NaN

[10 rows x 9 columns]

In [9]:
df2.tail(10)

UNITID  ... MN_EARN_WNE_P10
7102  48154401  ...             NaN
7103  48305401  ...             NaN
7104  48312401  ...             NaN
7105  48312402  ...             NaN
7106  48312403  ...             NaN
7107  48312404  ...             NaN
7108  48387801  ...             NaN
7109  48400201  ...             NaN
7110  48778201  ...             NaN
7111  49005401  ...             NaN

[10 rows x 9 columns]

In [10]:
df2.sample(10)

UNITID  ... MN_EARN_WNE_P10
2148    187505  ...             NaN
5938    470296  ...             NaN
5358    451121  ...             NaN
1760    173647  ...             NaN
690     133960  ...             NaN
7048  45133401  ...             NaN
2734    204200  ...             NaN
6217    482635  ...             NaN
5629    457785  ...             NaN
4834    439118  ...             NaN

[10 rows x 9 columns]

In [11]:
# Search for schools in the Baltimore area to look for UMBC, Johns Hopkins, and Loyola
# Loyola is called Loyola University Maryland instead of Loyola College Maryland
df2[df2["CITY"]== "Baltimore"]

UNITID  ... MN_EARN_WNE_P10
1451    161855  ...             NaN
1452    161864  ...             NaN
1453    161873  ...             NaN
1460    162283  ...             NaN
1462    162371  ...             NaN
1466    162654  ...             NaN
1473    162928  ...             NaN
1475    163046  ...             NaN
1477    163259  ...             NaN
1478    163268  ...             NaN
1480    163295  ...             NaN
1485    163453  ...             NaN
1487    163532  ...             NaN
1489    163578  ...             NaN
1491    163736  ...             NaN
1493    163815  ...             NaN
3004    212452  ...             NaN
4225    373784  ...             NaN
4762    434672  ...             NaN
5314    450076  ...             NaN
5834    461379  ...             NaN
6016    476601  ...             NaN
7006  43018403  ...             NaN

[23 rows x 9 columns]

In [12]:
UMBC = df2[df2["ZIP"]== "21250"]
UMBC

UNITID  ... MN_EARN_WNE_P10
1478  163268  ...             NaN

[1 rows x 9 columns]

In [13]:
UMCP = df2[df2["CITY"]== "College Park"]
UMCP

UNITID  ... MN_EARN_WNE_P10
1479  163286  ...             NaN

[1 rows x 9 columns]

In [14]:
JHU = df2[df2['INSTNM']== "Johns Hopkins University"]
JHU

UNITID                    INSTNM  ... TUITIONFEE_IN MN_EARN_WNE_P10
1473  162928  Johns Hopkins University  ...       52170.0             NaN

[1 rows x 9 columns]

In [15]:
# There is no Loyola College Maryland
df2[df2["INSTNM"]== "Loyola College Maryland"]

Empty DataFrame
Columns: [UNITID, INSTNM, CITY, STABBR, ZIP, ADM_RATE, UGDS, TUITIONFEE_IN, MN_EARN_WNE_P10]
Index: []

In [16]:
Loyola = df2[df2["INSTNM"]== "Loyola University Maryland"]
Loyola

UNITID                      INSTNM  ... TUITIONFEE_IN MN_EARN_WNE_P10
1475  163046  Loyola University Maryland  ...       47560.0             NaN

[1 rows x 9 columns]

In [17]:
# UMBC, Loyola, UMCP and Johns Hopkins are group together
df3 = df2[(df2['UNITID'].isin(['163046','162928', "163286", "163268"]))]
df3


UNITID  ... MN_EARN_WNE_P10
1473  162928  ...             NaN
1475  163046  ...             NaN
1478  163268  ...             NaN
1479  163286  ...             NaN

[4 rows x 9 columns]

In [18]:
# This calculates the min, max, and average in state tuition
df3['TUITIONFEE_IN'].describe()

count        4.000000
mean     30411.750000
std      22546.011271
min      10399.000000
25%      11238.250000
50%      29539.000000
75%      48712.500000
max      52170.000000
Name: TUITIONFEE_IN, dtype: float64

In [19]:
# Filter the smaller dataset further to only keep colleges from Maryland, 
# Virgina, Washington DC, Delaware, and Pennsylvinia.
Five_States_Colleges = df2[(df2['STABBR'].isin(['DE','DC', "VA", "MD", "PA"]))]
Five_States_Colleges

UNITID  ... MN_EARN_WNE_P10
634     130837  ...             NaN
635     130873  ...             NaN
636     130907  ...             NaN
637     130934  ...             NaN
638     130943  ...             NaN
...        ...  ...             ...
7063  45794301  ...             NaN
7084  47503303  ...             NaN
7089  47503308  ...             NaN
7094  47503313  ...             NaN
7103  48305401  ...             NaN

[690 rows x 9 columns]

In [20]:
type(Five_States_Colleges)

pandas.core.frame.DataFrame

### Save the filtered dataset to a new file named "Five_States_Colleges.csv" for future analysis. Upload this file to your GitHub along with the notebook.


In [21]:
FSC = Five_States_Colleges.to_csv('Five_State_Colleges.csv', index = True)
